### PR TITLE
summary of retry changes

### PR DESCRIPTION
**Pull request by ProfOak**
comments and small cleanups

added retry functionality for failed downloads

added logging statements

@NCI-GDC/ucdevs r?

### CODE REVIEW

File: gdc_client/download/parser.py [hunk 1]
```diff
@@ -75,7 +82,48 @@ def download(parser, args):
         ids.add(i['id'])
 
     client = get_client(args, args.token_file)
-    client.download_files(ids)
+    downloaded_files, errors = client.download_files(ids)
+
+    if args.retry_amount > 0:
+        files_not_downloaded = []
+
+        for uuid, _ in errors.items():
```
Comment (dankolbman) [inline gdc_client/download/parser.py]: same as `for uuid in errors.keys()`?

File: gdc_client/download/parser.py [hunk 2]
```diff
@@ -75,7 +82,48 @@ def download(parser, args):
...
+
+
+def retry_download(client, uuid, retry_amount, no_auto_retry):
+
+    log.info('Retrying download {}'.format(uuid))
+
+    e = True
+    while 0 < retry_amount and e:
+        if no_auto_retry:
+            should_retry = raw_input('Retry download for {}? (y/N): '.format(uuid))
```
Comment (dankolbman) [inline gdc_client/download/parser.py]: Don't know how I feel about accepting user input here. Can it just skip if the arg is set?

File: gdc_client/download/parser.py (hunk 1 above)
Comment (ProfOak) [inline gdc_client/download/parser.py]: Haha yeah I guess you're right. I think I was using the values before but scrapped it. Good eye.

File: gdc_client/download/parser.py (hunk 2 above)
Comment (ProfOak) [inline gdc_client/download/parser.py]: https://jira.opensciencedatacloud.org/browse/DTT-23 
This calls for a prompt. If you want to change that we might need more people's input on this.

Comment (dankolbman) [inline gdc_client/download/parser.py]: oh ok, I see that you can still have it run automatically without retrying with `--retry-amount 0`.

File: gdc_client/download/parser.py [hunk 3]
```diff
@@ -75,7 +82,48 @@ def download(parser, args):
...
+        else:
+            should_retry = 'y'
+
+        if should_retry.lower() == 'y':
+            log.info('{} retries remaning...'.format(retry_amount))
+            log.info('Retrying download... {} in 5 seconds'.format(uuid))
+            retry_amount -= 1
+            time.sleep(5)
```
Comment (madopal) [inline gdc_client/download/parser.py]: Make the timeout a const

**Comment by madopal** [comment]
I'm good assuming we make the wait for retry into a const or command line arg. I'm assuming we don't really know how long to wait, so it'd be good to be able to change it on the fly.

**Comment by madopal** [comment]
Great, lgtm as well.

**Pull request merged**